cmd/deploy.go

```diff
@@ -100,6 +104,8 @@ func newDeployCmd() *cobra.Command {
 	f.StringVarP(&dc.location, "location", "l", "", "location to deploy to (required)")
 	f.BoolVarP(&dc.forceOverwrite, "force-overwrite", "f", false, "automatically overwrite existing files in the output directory")
 	f.StringArrayVar(&dc.set, "set", []string{}, "set values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)")
+	f.BoolVar(&dc.useBuiltinAPIModel, "with-builtin-model", false, "use builtin api model, see --builtin-model-name to specify a specific builtin model")
+	f.StringVar(&dc.builtinAPIModelName, "builtin-model-name", defaultBuiltinAPIModelName, "set the builtin api model with the specified name")
```

Comment:
Why not use use one flag for the apimodel name and determine "use built-in model" by checking if that string value is empty?

Reply:
Thought about that. Mainly so the user doesn't have to supply the model name if they don't want to.... but not tied to this as is.

Also considered having `--api-model` look for a `builtin` value, with the tradeoff that the user can't really customize the model then (not that they can with what I have here, but there's the potential to).

Reply:
In that case, I have a different question: why let the user specify an apimodel if the point of the feature is to allow a quick / no customization deploy? If the user wants to specify the apimodel they can always do `acs-engine deploy --api-model pkg/api/models/v0.json`. I liked @khenidak's suggestion to have a "quick-create" option, instead of making it a command it could be `acs-engine deploy --quick-create […]`. Thoughts?

Reply:
Thinking about this a bit more, we could just use `--api-model` and then have a separate command to dump a built-in model that the user can then manually tweak.

Reply:
>  why let the user specify an apimodel

I think there's really two goals, a quick way to get started with blessed defaults, and a good way to make sure users are using the current best practices even if they don't necessarily know about some new setting.

Reply:
I think this needs to be deliberate by user. Just a thought

Reply:
Deliberate in what way?

Reply:
Discussed with @khenidak, updated to have just one flag.